client/src/components/FilterResult.jsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react'
+
+export default function FilterResult() {
+//getting filter result from url 
+let category = document.location.pathname.replace('/ExistingEntry/filter/','')
+if(category.replace('%20', " ")){
+    category=category.replace('%20', " ")
+}
+
+//fetching result
+
+console.log("inside")
+fetch(`/categories/${category}`)
+.then((res) => res.json())
+//if there is no matching
+.then((entry) => {
+    if(entry.length === 0){
+        document.getElementById("full-list").innerText="Didn't find any matching records"
+    }
+  else  {entry.forEach((element) => {
+      // container
+      let Cont = document.createElement("div");
+      Cont.id = element._id;
+      Cont.className = "listing-cont";
+
+      //title element
+      let Title = document.createElement("h3");
+      Title.innerText = element.title;
+      Title.className = "listing-title";
+
+      //author element
+      let Author = document.createElement("h5");
+      Author.innerText = element.author;
+      Author.className = "listing-author";
+
+      //text area element
+      let TextArea = document.createElement("h5");
+      TextArea.innerText = element.textArea;
+      TextArea.className = "listing-textArea";
+
+      //Category element
+      let Category = document.createElement("h5");
+      Category.innerText = element.category;
+      Category.className = "listing-category " + element.category;
+
+      //Date element
+      let Dates = document.createElement("h5");
+      Dates.innerText = element.date;
+      Dates.className = "listing-date";
+
+      //Edit button
+      let Edit = document.createElement("button");
+      Edit.innerText = "Edit";
+      Edit.className = "listing-edit";
+      Edit.id = element._id;
+      Edit.onclick = Editing;
+
+      //Delete button
+      let Delete = document.createElement("button");
+      Delete.innerText = "Delete";
+      Delete.className = "listing-delete";
+      Delete.onclick = Deleting;
+
+      //adding all elements to the container
+      Cont.appendChild(Edit);
+      Cont.appendChild(Delete);
+      Cont.appendChild(Title);
+      Cont.appendChild(Author);
+      Cont.appendChild(TextArea);
+      Cont.appendChild(Category);
+      Cont.appendChild(Dates);
+      //adding the container to the page
+      document.getElementById("full-list").appendChild(Cont);
```

Comment:
This is another spot that we can write straight into JSX. It would help your code be more understandable as well as less code overall